address-autocomplete/src/aem-cf-editor-1/web-src/src/components/AddressAutocompleteField.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * <license header>
+ */
+
+import React, { useEffect, useState, useRef } from "react";
+import { attach } from '@adobe/uix-guest';
+import { extensionId } from './Constants';
+import { lightTheme, TextField, Provider, ComboBox, Item } from '@adobe/react-spectrum';
+import { useAsyncList } from 'react-stately';
+
+export default () => {
+    const [connection, setConnection] = useState();
+    const [dataApi, setDataApi] = useState();
+    const inputRef = useRef();
+
+    useEffect(() => {
+        const init = async () => {
+            const connection = await attach({ id: extensionId });
+            setConnection(connection);
+
+            const dataApi = await connection.host.dataApi.get();
+            setDataApi(dataApi);
+
+            const autocomplete = new window.google.maps.places.Autocomplete(inputRef.current.getInputElement());
+            autocomplete.addListener('place_changed', async () => {
+                const place = autocomplete.getPlace();
+
+                if (!place.geometry) {
+                    console.error('No details available for input: ' + place.name);
+                    return;
+                }
+
+                const addressObject = {
+                    address: place.formatted_address,
+                    city: place.address_components.find(component => component.types.includes('locality')).long_name,
+                    state: place.address_components.find(component => component.types.includes('administrative_area_level_1')).short_name,
+                    postalCode: place.address_components.find(component => component.types.includes('postal_code')).short_name
+                };
+                console.log(addressObject);
+
+                await dataApi.setValue("address_autocomplete", addressObject.address);
+                await dataApi.setValue("city", addressObject.city);
+                await dataApi.setValue("state", addressObject.state);
+                await dataApi.setValue("postcode", addressObject.postalCode);
+            });
+
+            // workaround
+            await connection.host.field.setHeight(270);
+        };
+        init().catch(console.error);
+    }, []);
+
+    const list = useAsyncList({
+        async load({ signal, cursor, filterText }) {
+            if (!filterText) {
+                return;
+            }
+
+            const options = [
```

Comment:
Why are these options hardcoded?

Reply:
I pushed updated versions:
- The logic of working with the Google API has been extracted into a separate service.
- The variables have been moved to the environment (env)